### ISSUE TITLE
Docker.qcow2 has unexpected errors: either repair offline or reset to factory defaults

### ISSUE THREAD

**Issue by alexbeletsky** [docker/for-mac#1130]
### Expected behavior

Docker restarts and works.

### Actual behavior

Docker hands on restart.

### Information

After the latest upgrade I had a few of containers running. After a night any of containers responded, so I restarted docker via `Docker restart` UI menu item. 

<img width="326" alt="screenshot 2017-01-12 12 29 31" src="https://cloud.githubusercontent.com/assets/304929/21888049/cb3c853a-d8c2-11e6-8bc5-e94bd49e42cd.png">

The restart hangs.

Diagnostic ID: D0CA9B5C-A9BA-4373-8441-13EEE9FB4CF3

```plain
Docker for Mac: version: 1.13.0-rc5-beta35 (25211e84d)
OS X: version 10.11.6 (build: 15G1004)
logs: /tmp/D0CA9B5C-A9BA-4373-8441-13EEE9FB4CF3/20170112-121956.tar.gz
failure: Docker.qcow2 has unexpected errors: either repair offline or reset to factory defaults
[OK]     vmnetd
[OK]     dns
[OK]     driver.amd64-linux
[OK]     virtualization VT-X
[OK]     app
[OK]     moby
[OK]     system
[OK]     moby-syslog
[OK]     db
[OK]     env
[OK]     virtualization kern.hv_support
[OK]     slirp
[OK]     osxfs
[OK]     moby-console
[OK]     logs
[ERROR]  docker-cli
         docker ps failed
[OK]     menubar
[ERROR]  disk
         Docker.qcow2 has unexpected errors: either repair offline or reset to factory defaults
```

### Steps to reproduce the behavior

  1. ...
  2. ...

**Comment by djs55** [comment]
I think the `Docker.qcow2` error is a false positive -- there's no sign of block device problems in the logs. This can happen because `qemu-img check` assumes different invariants about the qcow2 file; in particular we run with "lazy refcounting" which seems to confuse it

I see other interesting things in the logs:

The hypervisor is asked to shutdown:
```
2017-01-12 12:13:06+01 ... Docker[com.docker.driver.amd64-linux][1514] <Notice>: sending SIGTERM to com.docker.hyperkit pid 1519
2017-01-12 12:13:06+01 ... Docker[com.docker.osx.hyperkit.linux][613] <Notice>: VM shutdown at 2017-01-12 12:13:06 +0100 CET
```
This just means that the shutdown signal has been sent. The VM still appears to be running and the console has:
```
/ # ESC[6nHangcheck: hangcheck value past margin!
Hangcheck: hangcheck value past margin!
 * Stopping local ... [ ok ]
 * Saving random seed ... [ ok ]
 * Stopping Hyper-V daemon: hv_vss_daemon ... * start-stop-daemon: no matching processes found
 [ ok ]
 * Stopping Hyper-V daemon: hv_kvp_daemon ... * start-stop-daemon: no matching processes found
 [ ok ]
 * Stopping busybox crond ... [ ok ]
 * Stopping chronyd ... [ ok ]
 * Stopping docker
 * start-stop-daemon: 1 process refused to stop
 * Stopping proxy port service ... [ ok ]
 * Stopping DHCP Client Daemon ... [ ok ]
 * sysklogd -> stop: klogd ... [ ok ]
 * sysklogd -> stop: syslogd ... [ ok ]
 * Stopping docker socket passthrough ... [ ok ]
 * Stopping FUSE socket passthrough ... * start-stop-daemon: 1 process refused to stop
 * Failed to stop transfused
 [ !! ]
 * ERROR: transfused failed to stop
 * Stopping busybox acpid ... [ ok ]
 * Unmounting loop devices
 * Unmounting filesystems
 *   Unmounting /var/lib/docker/aufs/mnt/d91d3209db4107f6acdb380417afb05def05dd9bc7028812de3a9b7b0a273244 ... [ ok ]
 *   Unmounting /var/lib/docker/aufs/mnt/b7dc1140e8e1fe16696893bf5254711eaf9c80ba120040cdcc329c9314aa166b ... [ ok ]
 *   Unmounting /var/lib/docker/aufs/mnt/c8cc0104892a4cf47f66bf28f98c55ed5128ed464a8e2b9fc7036fcd38a9c1c0 ... [ ok ]
 *   Unmounting /var/lib/docker/aufs/mnt/7273dd829b7a9023d1aba39d6691b5cb81445fb5a3dc376ed993abecb1515449 ... [ ok ]
 *   Unmounting /var/lib/docker/aufs/mnt/cea556ffb4881557cfb10bddab5614a61293a72d57c645ef1516c77d331d1d3a ... [ ok ]
 *   Unmounting /var/lib/docker/aufs/mnt/4886ec403533b8096968f593f7b45f3c736cf3173456d2116f51ce3011db9179 ... [ ok ]
 *   Unmounting /var/lib/docker/aufs/mnt/6d3f8d98324b3356755fda01053bc4450c1802e555ebadb441cd6d39a6dc4f94 ... [ ok ]
 *   Unmounting /var/lib/docker/aufs/mnt/e76c96d6039e5a2c988a9e0329c8dd56c94fe4a7b1e9378ff3305f8117fa8a09 ... [ ok ]
 *   Unmounting /var/lib/docker/aufs/mnt/4eabb01d520db5601179158b60b79ba5b04b0138084ab3b346926951257ea0c3 ... [ ok ]
 *   Unmounting /var/lib/docker/aufs/mnt/f7f47895706d911d2d128cb7c8a1b304d52dffc76e9a75bb07aa7bb555dae110 ... [ ok ]
 *   Unmounting /var/lib/docker/aufs ... [ ok ]
 *   Unmounting /port ... [ ok ]
 *   Unmounting /Database ... [ ok ]
unregister_netdevice: waiting for lo to become free. Usage count = 2
unregister_netdevice: waiting for lo to become free. Usage count = 2
unregister_netdevice: waiting for lo to become free. Usage count = 2
```
It looks like the VM shutdown wedged but for some reason we didn't send it a `SIGKILL`.

**Comment by djs55** [comment]
FWIW if you kill this process:
```
 1519   ??  S    108:16.68 /Applications/Docker.app/Contents/MacOS/com.docker.hyperkit
```
it should complete the shutdown and hopefully restart.

**Comment by alexbeletsky** [comment]
Thanks @djs55 .. I did reset to factory settings and had to rebuild all images, it works now, but hopefully diagnostic information could help to find a bug.

**Comment by djs55** [comment]
I think I can see the problem: the button in the UI does trigger a VM shutdown but it accidentally bypasses the code which sets a timeout and follows up with `SIGKILL`. I'll investigate in more detail. Thanks for the report!

**Comment by alexbeletsky** [comment]
@djs55 you are welcome, but besides that probem I have to admit 1.13.0-rc5-beta35 completely destroyed my development environment. I used to run 8-12 containers in parallel, serving different services.

Now the startup time of applications in containers increased and sometimes it's crashing without any clean reason. 

Is that possible to revert to previous version?

**Comment by dsheets** [comment]
@alexbeletsky are you still having the problems you described with Beta 35? Do they persist in Beta 37? If so, could you please file a new issue with more details about what you are running, what you expect to occur, and what you observe?

I'm marking this issue as fixed as @djs55 has prepared a patch but it is still under test to work out any regressions. Once that patch ships, we will close this issue as completed.

**Comment by alexbeletsky** [comment]
@dsheets currently on Version 1.13.0-beta39 (15140), all seems to be fine!
